### ISSUE TITLE
BedReady faild to install

### ISSUE THREAD

**Issue by piechade** [fredrikbaberg/hassio-addons#275]
Corresponding issue of the plugin: https://github.com/jneilliii/OctoPrint-BedReady/issues/3

```
Installiere Plugin "OctoPrint-BedReady" von https://github.com/jneilliii/OctoPrint-BedReady/archive/master.zip...
/data/python/octoprint/bin/python -m pip --disable-pip-version-check install file:///tmp/tmpl43d_949/OctoPrint-BedReady-master.zip --no-cache-dir
Processing /tmp/tmpl43d_949/OctoPrint-BedReady-master.zip
Preparing metadata (setup.py): started
Preparing metadata (setup.py): finished with status 'done'
Requirement already satisfied: OctoPrint in /data/python/octoprint/lib/python3.9/site-packages (from Bed-Ready==0.1.3) (1.7.3)
Collecting numpy==1.21.4
Downloading numpy-1.21.4.zip (10.6 MB)
Installing build dependencies: started
Installing build dependencies: finished with status 'done'
Getting requirements to build wheel: started
Getting requirements to build wheel: finished with status 'done'
Preparing metadata (pyproject.toml): started
Preparing metadata (pyproject.toml): finished with status 'done'
Collecting opencv-python-headless
Downloading opencv-python-headless-4.5.5.64.tar.gz (89.9 MB)
Installing build dependencies: started
Installing build dependencies: still running...
Installing build dependencies: still running...
Installing build dependencies: still running...
Installing build dependencies: finished with status 'done'
Getting requirements to build wheel: started
Getting requirements to build wheel: finished with status 'done'
Preparing metadata (pyproject.toml): started
Preparing metadata (pyproject.toml): finished with status 'done'
Requirement already satisfied: PyYAML<6,>=5.4.1 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (5.4.1)
Requirement already satisfied: watchdog==0.10.4 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (0.10.4)
Requirement already satisfied: OctoPrint-FirmwareCheck>=2021.10.11 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (2021.10.11)
Requirement already satisfied: Flask-Assets<3,>=2.0 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (2.0)
Requirement already satisfied: pkginfo<2,>=1.7.1 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (1.8.1)
Requirement already satisfied: pathvalidate<3,>=2.4.1 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (2.5.0)
Requirement already satisfied: Click<8,>=7.1.2 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (7.1.2)
Requirement already satisfied: netifaces<1,>=0.11 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (0.11.0)
Requirement already satisfied: zeroconf<0.34,>=0.33 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (0.33.4)
Requirement already satisfied: semantic-version<3,>=2.8.5 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (2.8.5)
Requirement already satisfied: Jinja2<3,>=2.11.3 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (2.11.3)
Requirement already satisfied: zipstream-new<1.2,>=1.1.8 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (1.1.8)
Requirement already satisfied: wrapt<1.13,>=1.12.1 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (1.12.1)
Requirement already satisfied: OctoPrint-PiSupport>=2021.10.28 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (2021.10.28)
Requirement already satisfied: feedparser<7,>=6.0.8 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (6.0.8)
Requirement already satisfied: markdown<3.2,>=3.1 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (3.1.1)
Requirement already satisfied: OctoPrint-FileCheck>=2021.2.23 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (2021.2.23)
Requirement already satisfied: websocket-client<1,>=0.59 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (0.59.0)
Requirement already satisfied: Flask-Login<0.6,>=0.5 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (0.5.0)
Requirement already satisfied: emoji<2,>=1.4.2 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (1.6.1)
Requirement already satisfied: psutil<6,>=5.8 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (5.8.0)
Requirement already satisfied: flask<2,>=1.1.4 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (1.1.4)
Requirement already satisfied: filetype<2,>=1.0.7 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (1.0.8)
Requirement already satisfied: werkzeug<2,>=1.0.1 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (1.0.1)
Requirement already satisfied: colorlog<6,>=5.0.1 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (5.0.1)
Requirement already satisfied: Flask-Babel<2,>=1.0 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (1.0.0)
Requirement already satisfied: regex in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (2021.11.10)
Requirement already satisfied: pyserial<4,>=3.4 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (3.5)
Requirement already satisfied: tornado<7,>=6 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (6.1)
Requirement already satisfied: markupsafe<2.0,>=1.1 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (1.1.1)
Requirement already satisfied: requests<3,>=2.26.0 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (2.26.0)
Requirement already satisfied: blinker<2,>=1.4 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (1.4)
Requirement already satisfied: pylru<2,>=1.2 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (1.2.0)
Requirement already satisfied: sentry-sdk<2,>=1.3.1 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (1.5.0)
Requirement already satisfied: immutabledict<3,>=2.1 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (2.2.1)
Requirement already satisfied: future<1,>=0.18.2 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (0.18.2)
Requirement already satisfied: sarge==0.1.6 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (0.1.6)
Requirement already satisfied: netaddr<0.9,>=0.8 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (0.8.0)
Requirement already satisfied: cachelib<0.2,>=0.1 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (0.1.1)
Requirement already satisfied: itsdangerous<2,>=1.1.0 in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (1.1.0)
Requirement already satisfied: unidecode in /data/python/octoprint/lib/python3.9/site-packages (from OctoPrint->Bed-Ready==0.1.3) (1.3.2)
Requirement already satisfied: pathtools>=0.1.1 in /data/python/octoprint/lib/python3.9/site-packages (from watchdog==0.10.4->OctoPrint->Bed-Ready==0.1.3) (0.1.2)
Requirement already satisfied: sgmllib3k in /data/python/octoprint/lib/python3.9/site-packages (from feedparser<7,>=6.0.8->OctoPrint->Bed-Ready==0.1.3) (1.0.0)
Requirement already satisfied: webassets>=2.0 in /data/python/octoprint/lib/python3.9/site-packages (from Flask-Assets<3,>=2.0->OctoPrint->Bed-Ready==0.1.3) (2.0)
Requirement already satisfied: pytz in /data/python/octoprint/lib/python3.9/site-packages (from Flask-Babel<2,>=1.0->OctoPrint->Bed-Ready==0.1.3) (2021.3)
Requirement already satisfied: Babel>=2.3 in /data/python/octoprint/lib/python3.9/site-packages (from Flask-Babel<2,>=1.0->OctoPrint->Bed-Ready==0.1.3) (2.9.1)
Requirement already satisfied: setuptools>=36 in /data/python/octoprint/lib/python3.9/site-packages (from markdown<3.2,>=3.1->OctoPrint->Bed-Ready==0.1.3) (52.0.0)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /data/python/octoprint/lib/python3.9/site-packages (from requests<3,>=2.26.0->OctoPrint->Bed-Ready==0.1.3) (1.26.7)
Requirement already satisfied: charset-normalizer~=2.0.0 in /data/python/octoprint/lib/python3.9/site-packages (from requests<3,>=2.26.0->OctoPrint->Bed-Ready==0.1.3) (2.0.7)
Requirement already satisfied: idna<4,>=2.5 in /data/python/octoprint/lib/python3.9/site-packages (from requests<3,>=2.26.0->OctoPrint->Bed-Ready==0.1.3) (3.3)
Requirement already satisfied: certifi>=2017.4.17 in /data/python/octoprint/lib/python3.9/site-packages (from requests<3,>=2.26.0->OctoPrint->Bed-Ready==0.1.3) (2021.10.8)
Requirement already satisfied: six in /data/python/octoprint/lib/python3.9/site-packages (from websocket-client<1,>=0.59->OctoPrint->Bed-Ready==0.1.3) (1.16.0)
Requirement already satisfied: ifaddr>=0.1.7 in /data/python/octoprint/lib/python3.9/site-packages (from zeroconf<0.34,>=0.33->OctoPrint->Bed-Ready==0.1.3) (0.1.7)
Building wheels for collected packages: Bed-Ready, numpy, opencv-python-headless
Building wheel for Bed-Ready (setup.py): started
Building wheel for Bed-Ready (setup.py): finished with status 'done'
Created wheel for Bed-Ready: filename=Bed_Ready-0.1.3-py2.py3-none-any.whl size=6808 sha256=1e0db6444a4720e36d156815c46d538e5e94fdf1c5e85de18c6c1caf0a13af47
Stored in directory: /tmp/pip-ephem-wheel-cache-_wdhi4k7/wheels/25/c8/fd/dc1ca875ee4512e9382ab7614c0702e862cf1d35fa1d0d2bae
Building wheel for numpy (pyproject.toml): started
Building wheel for numpy (pyproject.toml): still running...
Building wheel for numpy (pyproject.toml): still running...
Building wheel for numpy (pyproject.toml): still running...
Building wheel for numpy (pyproject.toml): still running...
Building wheel for numpy (pyproject.toml): still running...
Building wheel for numpy (pyproject.toml): finished with status 'done'
Created wheel for numpy: filename=numpy-1.21.4-cp39-cp39-linux_x86_64.whl size=22530003 sha256=440326bb2cda3ee6c05d27d55408d158be6ba6aaa2fdaed29d84b5ee18732e30
Stored in directory: /tmp/pip-ephem-wheel-cache-_wdhi4k7/wheels/14/97/c6/165e4deb0c59cfe0601822a5352d86f3ec79285d6429a07c1c
Building wheel for opencv-python-headless (pyproject.toml): started
ERROR: Command errored out with exit status 1:
command: /data/python/octoprint/bin/python /data/python/octoprint/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py build_wheel /tmp/tmp4d033c77
cwd: /tmp/pip-install-pvjmt0ma/opencv-python-headless_d5bd5fafcdea454bb46d2764338d8283
Complete output (9 lines):
File "/tmp/pip-build-env-h4eieuc3/overlay/lib/python3.9/site-packages/skbuild/setuptools_wrap.py", line 613, in setup
cmkr = cmaker.CMaker(cmake_executable)
File "/tmp/pip-build-env-h4eieuc3/overlay/lib/python3.9/site-packages/skbuild/cmaker.py", line 141, in __init__
self.cmake_version = get_cmake_version(self.cmake_executable)
File "/tmp/pip-build-env-h4eieuc3/overlay/lib/python3.9/site-packages/skbuild/cmaker.py", line 95, in get_cmake_version
raise SKBuildError(
Traceback (most recent call last):
Problem with the CMake installation, aborting build. CMake executable is cmake
----------------------------------------
ERROR: Failed building wheel for opencv-python-headless
ERROR: Could not build wheels for opencv-python-headless, which is required to install pyproject.toml-based projects
Building wheel for opencv-python-headless (pyproject.toml): finished with status 'error'
Successfully built Bed-Ready numpy
Failed to build opencv-python-headless
Fehler!
Could not parse output from pip, see plugin_pluginmanager_console.log for generated output
```

**Comment by fredrikbaberg** [comment]
I'll try to look at this over the weekend. What platform and architecture are you using?

**Comment by piechade** [comment]
Intel nuc 64bit

**Comment by fredrikbaberg** [comment]
I've confirmed the error occurs on my install, but haven't found a solution yet.

My plan has been to release an update to the addon after OctoPrint 1.8.0 is released. Since I haven't found an easy solution, I'll have to spend some time on figuring out how to proceed. There is a risk that I have to change base image from Alpine Linux to Debian. That's something I've decided against before, due to the increase in image size, but I'll have to look into that again.
